modules/assim.batch/R/pda.emulator.R

```diff
@@ -450,19 +487,27 @@ pda.emulator <- function(settings, params.id = NULL, param.names = NULL, prior.i
   
   for (c in seq_len(settings$assim.batch$chain)) {
     
-    m <- mcmc.out[[c]]$mcmc.samp
+    m <- matrix(NA, nrow =  nrow(mcmc.out[[c]]$mcmc.samp), ncol = length(prior.ind.all.ns))
     
       ## Set the prior functions back to work with actual parameter range
       
       prior.all <- do.call("rbind", prior.list)
       prior.fn.all <- pda.define.prior.fn(prior.all)
       
       ## Convert probabilities back to parameter values
-      for (i in seq_len(sum(n.param))) {
-        m[, i] <- eval(prior.fn.all$qprior[prior.ind.all][[i]], 
-                       list(p = mcmc.out[[c]]$mcmc.samp[, i]))
+      cc  <- 1
+      for (i in seq_along(prior.ind.all.ns)) {
+        # be careful that agrep(l) is working for now but it won't work for very similar parameter names
```

Comment:
Why are we using `agrep` here? I am extremely uncomfortable with approximate pattern matching in this context. If you're looking for partial matches, basic `grep` will already do that. If you know that the results will follow a pattern, it's much safer to construct an explicit regular expression pattern and use that.